doc/manual/performance-tips.rst

```diff
@@ -6,47 +6,126 @@
  Performance Tips
 ******************
 
-In the following sections, we briefly go through a few techniques that
-can help make your Julia code run as fast as possible.
+Many users are attracted to use Julia mainly because they were told Julia is fast. However, what they were not told is that why Julia is fast. Despite of the lack of this knowledge, they drafted some Julia code in a hurry, and ended up finding that it was even slower than the Python equivalent. Then, they became skeptic and complained it in the [Julia-user](https://groups.google.com/forum/#!forum/julia-users) mailing list. Generally, with the help of Julia team, their doubt got solved quickly.
 
-Avoid global variables
-----------------------
+This chapter describes how to write fast Julia code. If you concern about the speed of your Julia code, this chapter is what you should read over and over. Once you master the skills mentioned in this chapter, you are invited to improve this document.
+
+Why Julia *can* be fast
+-----------------
+
+The first thing is to understand why Julia *can* be fast. This is the basic to understand all following sections. Simply put, Julia is fast if and only if the binary generated by Julia compiler is efficient. While there are various compiling techniques to help generate efficient binary, the Julia compiler currently uses only one: *infer the runtime type*. And simply by employing this technique, Julia is already able to achieve nearly C/Fortran speed.
```

Comment:
If you're curious, @Sisyphuss, check out #3440. It's worth noting that some boxes not checked are already largely in place.